packages/mongodb-ts-autocomplete/src/index.ts

```diff
@@ -0,0 +1,256 @@
+import Autocompleter from '@mongodb-js/ts-autocomplete';
+import type {
+  AutocompleterOptions,
+  AutoCompletion,
+} from '@mongodb-js/ts-autocomplete';
+import { loadShellAPI } from './shell-api';
+import { loadBSONExpressions } from './bson-expressions';
+
+import type { JSONSchema } from './type-export';
+import { toTypescriptTypeDefinition } from './type-export';
+
+import {
+  extractPipelineUptoCaret,
+  extractPipelineFromLastAggregate,
+  inferCollectionNameFromFunctionCall,
+  inferMongoDBCommandFromFunctionCall,
+  isInAggregationPipelinePosition,
+  compileSourceFile,
+} from './cdt-analyser';
+
+import type { Pipeline, AutocompletionContext } from './autocompletion-context';
+
+type MongoDBAutocompleterOptions = {
+  context: AutocompletionContext;
+  autocompleterOptions?: AutocompleterOptions;
+};
+
+class DatabaseSchema {
+  private collectionNames: Set<string>;
+  private collectionSchemas: Record<string, JSONSchema>;
+
+  constructor() {
+    // TODO: this is kinda the only real reason for this class: So we can keep
+    // track of the known collectionNames for a database. It could be all the
+    // names from a listCollections() or it could just be all the ones we've
+    // auto-completed for. The schemas can come from the autocompletion context.
+    // This can probably be added to the autocompletion context.
+    this.collectionNames = new Set();
```

Comment:
Thinking about whether DatabaseSchema and ConnectionSchema can just be merged into the autocompletion context and then it keeps track of the known database names and for each database the known collection names.